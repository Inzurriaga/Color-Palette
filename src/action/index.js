export const toggleProjectDisplay  = () => ({
    type: "TOGGLE_PROJECT"
})

export const generateColorPalette = (palette) => ({
    type: "GENERATED_PALETTE",
    palette
})

export const toggleSaveModal = () => ({
    type: "TOGGLE_SAVE"
})

export const initialProjects = (projects) => ({
    type: "INITIAL_PROJECTS",
    projects
})

export const addNewProject = (project) => ({
    type: "ADD_PROJECT",
    project
})

export const initialPalette = (palettes) =>({
    type: "INITIAL_PALETTE",
    palettes
})

export const addNewPalette = (palette) => ({
    type: "ADD_PALETTE", 
    palette
})

export const deleteProject = (id) => ({
    type: "DELETE_PROJECT",
    id
})

export const deletePalette = (id) => ({
    type: "DELETE_Palette",
    id
})

export const deleteProjectPalettes = (projectId) => ({
    type: "DELETE_PROJECT_PALETTE",
    projectId
})