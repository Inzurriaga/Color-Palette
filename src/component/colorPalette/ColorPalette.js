import React, { Component } from "react";

export default class ColorPalette extends Component{
    constructor(){
        super();
        this.state = {
            colors: []
        }
    }

    componentDidMount(){
        this.generateColor()
    }

    generateColor = () => {
        let colorList = [];
        for(var i = 0; i < 5; i++){
            var randomColor = "#000000".replace(/0/g,function(){return (~~(Math.random()*16)).toString(16);});
            colorList.push(randomColor)
        };
        this.setState({
            colors: colorList
        })
    }

    render(){
        return(
            <div className="colorPalette-section">
                {
                    this.state.colors.map((color, index) => {
                        return(
                            <div className="color" id={index} style={{"background-color": color}}>
                                <p>{color}</p>
                            </div>
                        )
                    })
                }
            </div>
        )
    }
}