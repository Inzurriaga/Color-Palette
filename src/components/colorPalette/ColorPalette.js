import React, { Component } from "react";
import { connect } from "react-redux";
import { generateColorPalette, toggleLock } from "../../action"


export class ColorPalette extends Component{
    render(){
        return(
            <div className="colorPalette-section">
                {
                    this.props.colorPalette.map((color, index) => {
                        let hello;
                        if(color.lock){
                            hello = "lock"
                        }else{
                            hello = "not lock"
                        }
                        return(
                            <div className="color" id={index} key={index} style={{"backgroundColor": color.hex}}>
                                <button onClick={() => {this.props.toggleLock(index)}}>{hello}</button>
                                <p>{color.hex}</p>
                            </div>
                        )
                    })
                }
            </div>
        )
    }
}

export const mapStateToProps = (state) => ({
    colorPalette: state.colorPalette
})

export const mapDispatchToProps = (dispatch) => ({
    generateColorPalette: palette => dispatch(generateColorPalette(palette)),
    toggleLock: index => dispatch(toggleLock(index))
})

export default connect(mapStateToProps, mapDispatchToProps)(ColorPalette)