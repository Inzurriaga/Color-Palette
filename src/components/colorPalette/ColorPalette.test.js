import React from "react"
import { shallow } from "enzyme"
import { ColorPalette, mapDispatchToProps, mapStateToProps } from "./ColorPalette"
import { generateColorPalette, toggleLock } from "../../action"

const mockGenerateColorPalette = jest.fn()
const mockToggleLock = jest.fn()
const mockColorPalette = [{
    hex: "#000000",
    lock: true
},{
    hex: "#000000",
    lock: true
},{
    hex: "#000000",
    lock: true
},{
    hex: "#000000",
    lock: true
},{
    hex: "#000000",
    lock: true
}]

describe("App", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <ColorPalette generateColorPalette={mockGenerateColorPalette}
                        toggleLock={mockToggleLock}
                        colorPalette={mockColorPalette}/>
        )
    })
    describe("App", () => {
        it("should match snap shot", () => {
            expect(wrapper).toMatchSnapshot()
        })
        it("should invoke generateColorPalette when button is click", () => {

        })
        it("should invoke toggleLock when button is click", () => {
            
        })
    })
    describe("mapDispatchToProps", () => {
        it('calls dispatch with an generateInitialColorPalette action when called', () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = generateColorPalette(mockColorPalette)

            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.generateColorPalette(mockColorPalette)
      
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
          })
          it('calls dispatch with an getProjects action when called', () => {
            const mockDispatch = jest.fn()
            const actionToDispatch = toggleLock(1)
      
            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.toggleLock(1)
      
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
          })
    })
    describe("mapStateToProps", () => {
        it("should only return waht is needed from the store", () => {
            const mockStore = {
                colorPalette: [{},{},{}],
                project: [{},{},{}]
            }
            const expected = {
                colorPalette: [{},{},{}]
            }
            const result = mapStateToProps(mockStore)
            expect(result).toEqual(expected)
        })
    })
})