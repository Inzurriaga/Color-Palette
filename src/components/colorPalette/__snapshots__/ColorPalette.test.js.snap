// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should match snap shot 1`] = `
<div
  className="colorPalette-section"
>
  <div
    className="color"
    id={0}
    key="0"
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <button
      onClick={[Function]}
    >
      lock
    </button>
    <p>
      #000000
    </p>
  </div>
  <div
    className="color"
    id={1}
    key="1"
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <button
      onClick={[Function]}
    >
      lock
    </button>
    <p>
      #000000
    </p>
  </div>
  <div
    className="color"
    id={2}
    key="2"
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <button
      onClick={[Function]}
    >
      lock
    </button>
    <p>
      #000000
    </p>
  </div>
  <div
    className="color"
    id={3}
    key="3"
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <button
      onClick={[Function]}
    >
      lock
    </button>
    <p>
      #000000
    </p>
  </div>
  <div
    className="color"
    id={4}
    key="4"
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <button
      onClick={[Function]}
    >
      lock
    </button>
    <p>
      #000000
    </p>
  </div>
</div>
`;
